### PR TITLE
Feature/volley

### PR DESCRIPTION
**Pull request by KonradMiskiewicz**

### CODE REVIEW

File: app/src/main/java/com/example/teamyuml/decentworkmobile/AppSingleton.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import android.content.Context;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.toolbox.Volley;
+
+/*
+ * This class is responsible
+ * for configuration of Volley package
+ */
+public class AppSingleton {
```
Comment (KamilSupera) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/AppSingleton.java]: Why class name is AppSingleton?

File: app/src/main/java/com/example/teamyuml/decentworkmobile/AppSingleton.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    private static AppSingleton mAppSingletonInstance;
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/AppSingleton.java]: Add docstring to class fields pls cuz @KamilSupera or me won't know what these fields doing.

File: app/src/main/java/com/example/teamyuml/decentworkmobile/AppSingleton.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    private RequestQueue mRequestQueue;
```
Comment (KamilSupera) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/AppSingleton.java]: One style create fields. 
private 
private
private static

File: app/src/main/java/com/example/teamyuml/decentworkmobile/AppSingleton.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    private static Context mContext;
+
+    private AppSingleton(Context context) {
+        mContext = context;
+        mRequestQueue = getRequestQueue();
+    }
+
+    public static synchronized AppSingleton getInstance(Context context) {
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/AppSingleton.java]: First public methods than private methods.

File: app/src/main/java/com/example/teamyuml/decentworkmobile/AppSingleton.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+import android.content.Context;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.toolbox.Volley;
+
+/*
+ * This class is responsible
+ * for configuration of Volley package
+ */
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/AppSingleton.java]: I think this docstring is misguiding cuz it not configure volley. IMO this class is an instance for global AppSingleton object which let you makes http request to server and do not initialize it every single time you make request.

**Review by kaniak274 — APPROVED** [review]

File: app/src/main/java/com/example/teamyuml/decentworkmobile/VolleyInstance.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    public RequestQueue getRequestQueue() {
+        if (mRequestQueue == null) {
+            // getApplicationContext() is key, it keeps you from leaking the
+            // Activity or BroadcastReceiver if someone passes one in.
+            mRequestQueue = Volley.newRequestQueue(mContext.getApplicationContext());
+        }
+        return mRequestQueue;
+    }
+
+    public <T> void addToRequestQueue(Request<T> req,String tag) {
```
Comment (KamilSupera) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/VolleyInstance.java]: Missing space req, String

File: app/src/main/java/com/example/teamyuml/decentworkmobile/VolleyInstance.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        return mAppSingletonInstance;
+    }
+
+    public RequestQueue getRequestQueue() {
+        if (mRequestQueue == null) {
+            // getApplicationContext() is key, it keeps you from leaking the
+            // Activity or BroadcastReceiver if someone passes one in.
+            mRequestQueue = Volley.newRequestQueue(mContext.getApplicationContext());
+        }
+        return mRequestQueue;
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/VolleyInstance.java:34]: One line between if{} and return.

File: app/src/main/java/com/example/teamyuml/decentworkmobile/VolleyInstance.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    private RequestQueue mRequestQueue;
+    private static VolleyInstance mAppSingletonInstance;
+    private static Context mContext;
+
+
+    public static synchronized VolleyInstance getInstance(Context context) {
+        if (mAppSingletonInstance == null) {
+            mAppSingletonInstance = new VolleyInstance(context);
+        }
+        return mAppSingletonInstance;
```
Comment (kaniak274) [inline app/src/main/java/com/example/teamyuml/decentworkmobile/VolleyInstance.java:24]: One line between if and return.

**Pull request merged**